### PR TITLE
Add Player Argument Type for Brigadier

### PR DESCRIPTION
**Pull request by DevSolaris**
# Description
Added `ArgumentTypes.player()` and `ArgumentTypes.players()` as discussed a while ago in https://discord.com/channels/792754410576019477/809866076606693427/1396918734945521685

The error messages are the same as the ones you get from Paper
image

I am looking for some feedback on how I am asserting, as I don't personally like that they are in the command implementation.

All other ArgumentTypes are UOEs, if this PR is merged, I will also be implementing the other ones.

# Checklist
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

**Comment by 4everTheOne** [comment]
Hello @DevSolaris ! Thank you for your contribution! Overall the code looks good to me, just have a suggestion like you said/asked to improve the unit tests.
A suggestion for that would be to reduce the test scope for a unit test, something like this:
```java
	@Test
	void resolveAll() throws CommandSyntaxException
	{
		var reader = new StringReader("@a");

		var notch = serverMock.addPlayer("notch");
		var jeb = serverMock.addPlayer("jeb");
		var stack = CommandSourceStackMock.from(notch);

		var actual = ArgumentTypes.players().parse(reader, notch).resolve(stack);

		assertEquals(List.of(notch, jeb), actual);
	}

	@Test
	void resolveSelf() throws CommandSyntaxException
	{
		var reader = new StringReader("@s");

		var notch = serverMock.addPlayer("notch");
		var jeb = serverMock.addPlayer("jeb");
		var stack = CommandSourceStackMock.from(notch);

		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);

		assertEquals(List.of(notch), actual);
	}

	@Test
	void resolveRandom() throws CommandSyntaxException
	{
		var reader = new StringReader("@s");

		var notch = serverMock.addPlayer("notch");
		var jeb = serverMock.addPlayer("jeb");
		var stack = CommandSourceStackMock.from(notch);

		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);

		assertNotNull(actual);
		assertEquals(1, actual.size());
	}

	@Test
	void resolveByName() throws CommandSyntaxException
	{
		var reader = new StringReader("jeb");

		var notch = serverMock.addPlayer("notch");
		var jeb = serverMock.addPlayer("jeb");
		var stack = CommandSourceStackMock.from(notch);

		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);

		assertEquals(List.of(jeb), actual);
	}
```

With this you can also test the negative cases:
```java
	@Test
	void wrongUseOfPlayer() throws CommandSyntaxException
	{
		var reader = new StringReader("@a");

		var notch = serverMock.addPlayer("notch");
		var jeb = serverMock.addPlayer("jeb");
		var stack = CommandSourceStackMock.from(notch);

		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);

		assertEquals(Collections.emptyList(), actual);
		assertThat(notch, MessageTargetReceivedMessageMatcher.hasReceived("Only one player is allowed, but the provided selector allows more than one"));
	}
```

What do you think?

File: src/main/java/org/mockbukkit/mockbukkit/command/brigadier/argument/PlayerSelectorArgumentResolverMock.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		this.input = input;
+		this.single = single;
+	}
+
+	@Override
+	public @NotNull List<Player> resolve(@NotNull CommandSourceStack source)
+	{
+		if (single && input.equals("@a"))
+		{
+			String message = org.mockbukkit.mockbukkit.adventure.Languages.getInstance().getOrDefault("command.expected.a_single_player", "Only one player is allowed, but the provided selector allows more than one");
```
Comment (4everTheOne) [inline src/main/java/org/mockbukkit/mockbukkit/command/brigadier/argument/PlayerSelectorArgumentResolverMock.java]: Please replace the `org.mockbukkit.mockbukkit.adventure.Languages` with `Languages`. (Same on line 49)

**Comment by DevSolaris** [comment]
> Hello @DevSolaris ! Thank you for your contribution! Overall the code looks good to me, just have a suggestion like you said/asked to improve the unit tests. A suggestion for that would be to reduce the test scope for a unit test, something like this:
> 
> ```java
> 	@Test
> 	void resolveAll() throws CommandSyntaxException
> 	{
> 		var reader = new StringReader("@a");
> 
> 		var notch = serverMock.addPlayer("notch");
> 		var jeb = serverMock.addPlayer("jeb");
> 		var stack = CommandSourceStackMock.from(notch);
> 
> 		var actual = ArgumentTypes.players().parse(reader, notch).resolve(stack);
> 
> 		assertEquals(List.of(notch, jeb), actual);
> 	}
> 
> 	@Test
> 	void resolveSelf() throws CommandSyntaxException
> 	{
> 		var reader = new StringReader("@s");
> 
> 		var notch = serverMock.addPlayer("notch");
> 		var jeb = serverMock.addPlayer("jeb");
> 		var stack = CommandSourceStackMock.from(notch);
> 
> 		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);
> 
> 		assertEquals(List.of(notch), actual);
> 	}
> 
> 	@Test
> 	void resolveRandom() throws CommandSyntaxException
> 	{
> 		var reader = new StringReader("@s");
> 
> 		var notch = serverMock.addPlayer("notch");
> 		var jeb = serverMock.addPlayer("jeb");
> 		var stack = CommandSourceStackMock.from(notch);
> 
> 		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);
> 
> 		assertNotNull(actual);
> 		assertEquals(1, actual.size());
> 	}
> 
> 	@Test
> 	void resolveByName() throws CommandSyntaxException
> 	{
> 		var reader = new StringReader("jeb");
> 
> 		var notch = serverMock.addPlayer("notch");
> 		var jeb = serverMock.addPlayer("jeb");
> 		var stack = CommandSourceStackMock.from(notch);
> 
> 		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);
> 
> 		assertEquals(List.of(jeb), actual);
> 	}
> ```
> 
> With this you can also test the negative cases:
> 
> ```java
> 	@Test
> 	void wrongUseOfPlayer() throws CommandSyntaxException
> 	{
> 		var reader = new StringReader("@a");
> 
> 		var notch = serverMock.addPlayer("notch");
> 		var jeb = serverMock.addPlayer("jeb");
> 		var stack = CommandSourceStackMock.from(notch);
> 
> 		var actual = ArgumentTypes.player().parse(reader, notch).resolve(stack);
> 
> 		assertEquals(Collections.emptyList(), actual);
> 		assertThat(notch, MessageTargetReceivedMessageMatcher.hasReceived("Only one player is allowed, but the provided selector allows more than one"));
> 	}
> ```
> 
> What do you think?

Really nice suggestion c: thanks a lot!

I have pushed an update with your feedback and fixed the checkstyle error.

**Review by 4everTheOne — APPROVED** [review]
Thank you for this contribution!
